scipy/stats/tests/test_distributions.py

```diff
@@ -913,6 +914,93 @@ def test_sf(self):
         expected = (scale/x)**b   # 2.25e-18
         assert_allclose(p, expected)
 
+    @pytest.mark.filterwarnings("ignore:invalid value encountered in "
+                                "double_scalars")
+    @pytest.mark.parametrize("rvs_shape", [1, 2])
+    @pytest.mark.parametrize("rvs_loc", [0, 2])
+    @pytest.mark.parametrize("rvs_scale", [1, 5])
+    def test_fit(self, rvs_shape, rvs_loc, rvs_scale):
+        data = stats.pareto.rvs(size=100, b=rvs_shape, scale=rvs_scale,
+                                loc=rvs_loc)
+
+        # shape can still be fixed with multiple names
+        shape_mle1 = stats.pareto.fit(data, floc=0, f0=1.04)[0]
+        shape_mle2 = stats.pareto.fit(data, floc=0, fix_b=1.04)[0]
+        shape_mle3 = stats.pareto.fit(data, floc=0, fb=1.04)[0]
+        assert shape_mle1 == shape_mle2 == shape_mle3 == 1.04
+
+        # data can be shifted with changes to `loc`
+        data = stats.pareto.rvs(size=100, b=rvs_shape, scale=rvs_scale,
+                                loc=(rvs_loc + 2))
+        shape_mle, loc_mle, scale_mle = stats.pareto.fit(data, floc=2)
+        assert_equal(scale_mle + 2, data.min())
+        assert_equal(shape_mle, 1/((1/len(data - 2)) *
+                                   np.sum(np.log((data
+                                                  - 2)/(data.min() - 2)))))
+        assert_equal(loc_mle, 2)
+
+    @pytest.mark.filterwarnings("ignore:invalid value encountered in "
+                                "double_scalars")
+    @pytest.mark.parametrize("rvs_shape", [1, 2])
+    @pytest.mark.parametrize("rvs_loc", [0, 2])
+    @pytest.mark.parametrize("rvs_scale", [1, 5])
+    def test_fit_MLE_comp_optimzer(self, rvs_shape, rvs_loc, rvs_scale):
+        data = stats.pareto.rvs(size=100, b=rvs_shape, scale=rvs_scale,
+                                loc=rvs_loc)
+        args = [data, (stats.pareto._fitstart(data), )]
+        func = stats.pareto._reduce_func(args, {})[1]
+
+        # fixed `floc` to actual location provides as good or better fit.
+        mle = stats.pareto.fit(data, floc=rvs_loc)
+        opt = super(type(stats.pareto), stats.pareto).fit(data, floc=rvs_loc)
+        ll_mle = func(mle, data)
+        ll_opt = func(opt, data)
+        assert ll_mle < ll_opt or np.allclose(ll_mle, ll_opt)
```

Comment:
Would it help to break this out as a method
```
def test_fit_params(**kwargs):
    mle = stats.pareto.fit(data, **kwargs)
    opt = super(type(stats.pareto), stats.pareto).fit(data, **kwargs)
    ll_mle = func(mle, data)
    ll_opt = func(opt, data)
    assert ll_mle < ll_opt or np.allclose(ll_mle, ll_opt)
```
then these tests become
```
test_fit_params(floc=rvs_loc)
test_fit_params(floc=0)
test_fit_params(floc=0, f0=4)
test_fit_params(floc=0, fscale=rvs_scale/2)
```

Reply:
Great idea, I think we could even use it for other distributions as I suspect that it could be useful to be able to compare the log-likelihood results of other distributions. For example, in 

test_distributions.TestLaplace.test_fit_MLE_comp_optimzer we wrote:

```
        loc, scale = stats.laplace.fit(data)
        loc_opt, scale_opt = super(type(stats.laplace),
                                   stats.laplace).fit(data)
        ll_mle = ll(loc, scale, data)
        ll_opt = ll(loc_opt, scale_opt, data)
        assert ll_mle < ll_opt or np.allclose(ll_mle, ll_opt,
                                              atol=1e-15, rtol=1e-15)
```

But with this method we could easily check that fixing either variable with MLE gives a better LL result than numerical optimization as well as when no parameters are fixed. 

For now, I have left the signature as 
```
def _assert_lessthan_loglike(dist, data, func, **kwds):
```

while considering how to make this as general but still useful as possible.

Reply:
> Other considerations we might want:

Isn't what you have sufficiently general for now? Currently, I think we only want to test the difference between the super and overridden fit methods.